localstack/utils/docker_utils.py

```diff
@@ -282,10 +286,42 @@ def get_container_status(self, container_name: str) -> DockerContainerStatus:
         """Returns the status of the container with the given name"""
         pass
 
-    @abstractmethod
-    def get_network(self, container_name: str) -> str:
-        """Returns the network mode of the container with the given name"""
-        pass
+    def get_networks(self, container_name: str) -> List[str]:
+        LOG.debug("Getting networks for container: %s", container_name)
+        container_attrs = self.inspect_container(container_name_or_id=container_name)
+        return list(container_attrs["NetworkSettings"]["Networks"].keys())
+
+    def get_container_ipv4_for_network(
+        self, container_name_or_id: str, container_network: str
+    ) -> str:
+        """
+        Returns the IPv4 address for the container on the interface connected to the given network
+        :param container_name_or_id: Container to inspect
+        :param container_network: Network the IP address will belong to
+        :return: IP address of the given container on the interface connected to the given network
+        """
+        LOG.debug(
+            "Getting ipv4 address for container %s in network %s.",
+            container_name_or_id,
+            container_network,
+        )
+        # we always need the ID for this
+        container_id = self.get_container_id(container_name=container_name_or_id)
```

Comment:
nit: Wondering if we should add a heuristic to the implementation of `get_container_id(..)` and check if the string is a 64-chars long hex string (in which case it would be pretty safe to assume that the string contains an ID, and we could skip the conversion). Thoughts?

Reply:
It would potentially improve performance by a few ms. On the other hand, it is possible to create a container with a id-like string as name, so we cannot really check at this point if its an id or name. Might not be worth the trouble since the result of the method should be cached anyway if possible.